### PR TITLE
Update README.md and validate hyperlinks

### PR DESCRIPTION
**Pull request by wangkuiyi**
Fixes https://github.com/PaddlePaddle/Paddle/issues/1658

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -59,36 +59,36 @@ Please refer to our [release announcement](https://github.com/PaddlePaddle/Paddl
     the capability of PaddlePaddle to make a huge impact for your product.
 
 ## Installation
-Check out the [Install Guide](http://paddlepaddle.org/doc/build/) to install from
-pre-built packages (**docker image**, **deb package**) or
-directly build on **Linux** and **Mac OS X** from the source code.
+
+It is recommended to check out the
+[Docker installation guide](http://www.paddlepaddle.org/develop/doc/getstarted/build_and_install/docker_install_en.html)
```
Comment (typhoonzero) [inline README.md]: This guide uses a specified version tag `0.10.0rc1`, and the doc is not managed by version like "doc/0.10.0/...". So shall we use the `latest` tag instead.

Comment (wangkuiyi) [inline README.md]: Do you mean we should have `doc/<branch-or-tag>` for all branches and tags?  That would be great. I don't know how @reyoung created the auto build and deploy system. But he could have some ideas.

Comment (reyoung) [inline README.md]: 自动编译部署相关的代码在[这里](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/scripts/travis/docs.sh#L62)。

目前可以自定义任何分支的自动编译和部署。对于tag的自动编译和部署似乎并不必要，因为tag本身不是一个能够持续更新的东西，只会部署一次即可。
目前的目录结构是 `<branch>/doc` 和 `<branch>/doc_cn`。

Comment (typhoonzero) [inline README.md]: @reyoung 是考虑用户在使用对应版本的paddle可以找到对应版本的doc，我理解sphinx可以自动完成这件事情？[这里](https://github.com/PaddlePaddle/Paddle/blob/develop/paddle/scripts/travis/docs.sh#L62)看只有master和develop分支？

**Pull request merged**